server/aesdsocket.c

```diff
@@ -0,0 +1,276 @@
+	int getaddr,bind_status,listen_status; //Variables to get address for bind, return value of bind and listen
+	ssize_t rec_status=1; //Return of recv function
+	socklen_t size=sizeof(struct sockaddr); 
+	char buff[100]; //temporary buffer to store receiver data
+	int fd_status;
+	ssize_t write_status; //Retur of write function
+	int send_status; //Return of send function
+	int i, total_bytes=0,packet_bytes_total=0; //Store total bytes received in a packet and total bytes received in a connection
+	bool write_flag=false; 
+	
+	openlog(NULL,LOG_PID, LOG_USER); //To setup logging with LOG_USER
+	
+	
+	//To start a daemon process
+	if((argc>1) && strcmp(argv[1],"-d")==0)
+	{
+		if(daemon(0,0)==-1)
+		{
+			syslog(LOG_ERR, "Couldn't enter daemon mode");
+			exit(1);
+		}
+	}
+	
+	//To check if a signal is received
+	if(signal(SIGINT,signal_handler)==SIG_ERR)
+	{
+		syslog(LOG_ERR,"SIGINT failed");
+		exit(2);
+	}
+	if(signal(SIGTERM,signal_handler)==SIG_ERR)
+	{
+		syslog(LOG_ERR,"SIGTERM failed");
+		exit(3);
+	}
+	
+	//create socket
+	socket_fd=socket(PF_INET, SOCK_STREAM, 0);
+	if(socket_fd==-1)
+	{
+		syslog(LOG_ERR, "Couldn't create a socket");
+		exit(4);
+	}
+	
+	//Get server address
+	hints.ai_flags=AI_PASSIVE; //Set this flag before passing to function
+	getaddr=getaddrinfo(NULL,"9000",&hints,&servinfo);
+	if(getaddr !=0)
+	{
+		syslog(LOG_ERR, "Couldn't get the server's address");
+		exit(5);
+	}
+	
+	//Bind
+	bind_status=bind(socket_fd,servinfo->ai_addr,sizeof(struct sockaddr));
+	if(bind_status==-1)
+	{
+		freeaddrinfo(servinfo); //Free the memory before exiting
+		syslog(LOG_ERR, "Binding failed");
+		exit(6);
+	}
+	
+	freeaddrinfo(servinfo); //Free this memory before next operations
+	
+	fd_status=open("/var/tmp/aesdsocketdata.txt", O_WRONLY | O_CREAT | O_TRUNC, S_IRWXU | S_IRWXG | S_IRWXO); //To create a file and give permissions to all
+	if (fd_status==-1)
+	{
+		syslog(LOG_ERR, "The file could not be created/found");
+		exit(7);
+	}
+	close(fd_status);
+	
+	//Infinite loop for server-client connections
+	while(1)
+	{
+	
+	//Listen
+	listen_status=listen(socket_fd,20); //Max queue of connections set to 20
+	if(listen_status==-1)
+	{
+		syslog(LOG_ERR, "Listening to the connections failed");
+		exit(8);
+	}
+	
+	//accept
+	accept_return=accept(socket_fd,(struct sockaddr *)&client_addr,&size);
+	if(accept_return==-1)
+	{
+		syslog(LOG_ERR, "Connection could not be accepted");
+		exit(9);
+	}
+	else
+	{
+		syslog(LOG_INFO,"Accepts connection from %s",inet_ntoa(client_addr.sin_addr));
+		printf("Accepts connection from %s\n",inet_ntoa(client_addr.sin_addr));
+	}
+	
+	
+	store_data = (char*)malloc(100); //Allocate a memory
+	if(store_data==NULL)
+	{
+		syslog(LOG_ERR, "Memory couldn't be allocated");
+		exit(10);
+	}
+	
+	memset(store_data,0,100); //Reset the temporary buffer
+	
+	write_flag=false;
+	
+	//Loop to receive data packet
+	while(!write_flag)
+	{
+	
+	rec_status=recv(accept_return,buff,100,0); //Receive data packets from client
```

Comment:
@shashankchandrasekaran use a const int or #define for 100 rather than copying this value everywhere.